packages/synapse-react-client/src/components/QueryWrapper/useComputeRowSelectionPrimaryKey.ts

```diff
@@ -0,0 +1,35 @@
+import { useMemo } from 'react'
+import { getDefaultPrimaryKey } from '../SynapseTable/SynapseTableUtils'
+import { useGetEntity } from '../../synapse-queries'
+import { Table } from '@sage-bionetworks/synapse-types'
+import { useQuery } from '@tanstack/react-query'
+import { QueryContextType } from '../QueryContext'
+
+type UseComputeRowSelectionPrimaryKeyOptions = {
+  entityId: string
+  versionNumber?: number
+  rowSelectionPrimaryKeyFromProps?: string[]
+  queryMetadataQueryOptions: QueryContextType['queryMetadataQueryOptions']
+}
+
+export default function useComputeRowSelectionPrimaryKey(
```

Comment:
New custom hook with logic that was previously in QueryWrapper, extracted to make QueryWrapper as lean as possible.